lib/connect/ws.js

```diff
@@ -32,15 +40,14 @@ function buildBuilder (client, opts) {
     wsOpt.protocol = 'mqttv3.1'
   }
 
-  if (opts.protocol === 'wss') {
-    wssProperties.forEach(function (prop) {
-      if (opts.hasOwnProperty(prop)) {
-        wsOpt[prop] = opts[prop]
-      }
-    })
-  }
-
-  return websocket(url, wsOpt)
+  /*delete some mqtt overhead*/
+  mqttProperties.forEach(function (prop) {
+    if (opts.hasOwnProperty(prop)) {
+      delete opts[prop];
```

Comment:
It's bad anyway because wsOpt is a reference to opts and no "real" object. I changed this in a later commit in this branche. I'll do this because there are some options with the same key in mqtt and wss constructur (e.g. protocolVersion) 
If I don't delet them the websocket want to use this propertie too.